src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
 import './App.scss';
+import { Todo } from './type/Todo';
+import { User } from './type/User';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+
+function getTodos(): Todo[] {
+  return todosFromServer.map(todo => {
+    const copy: Todo = {
+      ...todo,
+      user: usersFromServer.find(user => user.id === todo.userId) || null,
+    };
+
+    return copy;
+  });
+}
+
+function getUserByName(fullName: string): User | null {
+  return usersFromServer.find(user => user.name === fullName) || null;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState(getTodos());
+  const [title, setTitle] = useState('');
+  const [fullName, setFullName] = useState('');
+  const [submitError, setSubmitError] = useState(false);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!fullName || !title) {
+      setSubmitError(true);
+
+      return;
+    }
+
+    if (submitError) {
+      setSubmitError(false);
+    }
+
+    setTodos(previous => {
+      const newTask: Todo = {
+        id: Math.max(...(previous.map(({ id }) => id))) + 1,
+        completed: false,
+        title,
+        user: getUserByName(fullName),
+      };
+
+      return [...previous, newTask];
+    });
+    setTitle('');
+    setFullName('');
+  };
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+  };
+
+  const handleChangeUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setFullName(value);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form action="/api/users" method="POST" onSubmit={handleSubmit}>
```

Comment:
```suggestion
      <form
           action="/api/users"
           method="POST"
           onSubmit={handleSubmit}
         >
```
if you have 3 and more atributes, better write on each line